screens/Trivia/index.js

```diff
@@ -8,41 +8,38 @@ import {
 } from 'react-native';
 import { HeaderBackButton } from '@react-navigation/stack';
 import { Text } from 'react-native-elements';
+import { useDispatch, useSelector } from 'react-redux';
 
 import Category from 'components/Category';
 import Difficulty from 'components/Difficulty';
-import { AppContext } from 'context';
+import { getTriviaData } from 'functions/quiz';
+import { fetchQuizCategories } from 'store/reducers/quizCategories';
 import { TRIVIAQ_SCREEN } from 'screens/routes';
 import colors from 'consts/colors';
 
 const Trivia = ({ navigation }) => {
-  const { categoryData, getAllCategoryData, getTriviaData } = useContext(
-    AppContext,
-  );
-
   const [difficulty, setDifficulty] = useState('');
   const [categoryId, setCategoryId] = useState();
   const [loading, setLoading] = useState(false);
   const [phaseTwo, setPhaseTwo] = useState(false);
+  const categories = useSelector((state) => state.quizCategories.categories);
+  const fetchingCategories = useSelector(
+    (state) => state.quizCategories.isFetching,
+  );
+  const dispatch = useDispatch();
 
   useEffect(() => {
-    getCategories();
-  }, [categoryData]);
-
-  const getCategories = async () => {
-    if (categoryData.length === 0) {
-      setLoading(true);
-      await getAllCategoryData();
-      setLoading(false);
+    if (categories.length === 0) {
+      dispatch(fetchQuizCategories());
     }
-  };
+  }, []);
 
   const handleSubmit = async () => {
     console.log(categoryId, difficulty);
     setLoading(true);
-    await getTriviaData(categoryId, difficulty);
+    const triviaData = await getTriviaData(categoryId, difficulty);
     setLoading(false);
-    navigation.navigate(TRIVIAQ_SCREEN);
+    navigation.navigate(TRIVIAQ_SCREEN, { triviaData });
   };
 
   React.useLayoutEffect(() => {
```

Comment:
I know this isn't a change to this PR but i've always wondered what the purpose of this hook really was under the hood. Why would you use this vs a useEffect?

Reply:
useLayoutEffect runs synchronously and useEffect asynchronously. This just ensures the button override takes. useEffect might work, would have to try - but I think from trying in previous projects I was getting inconsistencies. Here is a resource that explains the differences pretty well: https://daveceddia.com/useeffect-vs-uselayouteffect/